### ISSUE TITLE
Add delays to write operations (doesn't work if CPU is too fast)

### ISSUE THREAD

**Issue by kolen** [kellerkindt/pcd8544#4]
Currently, the library does not work if CPU is too fast. For me, on stm32f103 it fails on:

```rust
let clocks = rcc
    .cfgr
    .use_hse(8.mhz())
    .sysclk(72.mhz())
    .pclk1(36.mhz())
    .freeze(&mut flash.acr);
```

and works on:

```rust
let clocks = rcc
    .cfgr
    .use_hse(8.mhz())
    .sysclk(16.mhz())
    .pclk1(16.mhz())
    .freeze(&mut flash.acr);
```

According to [datasheet](https://www.sparkfun.com/datasheets/LCD/Monochrome/Nokia5110.pdf) (section 6.1.8), maximum clock speed is 4 Mbit/s.

I don't know yet what is proper way to do this, either [`embedded_hal::timer::CountDown`](https://docs.rs/embedded-hal/0.2.3/embedded_hal/timer/trait.CountDown.html) with `nb`'s [`block!`](https://docs.rs/nb/0.1.2/nb/macro.block.html), or something else. `bitbang-hal` [does it that way](https://github.com/sajattack/bitbang-hal/blob/f70f7f97fff28e0e21f72df9e09645677270b293/src/spi.rs#L123).

**Comment by kolen** [comment]
I was able to make it work independent of CPU speed by using hardware SPI instead of bit-banging: https://github.com/kolen/pcd8544/commit/6149845869383ef5f6be38bc461e21d7a01b0f4a (this commit is not based on current master and has few incompatible changes below it so I didn't yet create pull request for it).
SPI isn't mentioned in PCD8544 datasheet, but it's compatible with it in CPOL=0, CPHA=0 mode. (I used SPI without MISO line, as PCD8544 doesn't have read operations).

**Comment by kellerkindt** [comment]
Is there a way to provide both, the raw `OutputPin` access and the `SPI` approach? I guess not everyone wants to sacrifice an `SPI` port. Maybe by using an feature flag...?
When upgrading to the `v2::OutputPin` I also thought about introducing an separate error type, similar  to yours, in which every pin gets its own variant entry. Maybe that is a better approach than the current solution in 0.2.0

**Comment by kellerkindt** [comment]
Fixed by #7 - if not, feel free to open another issue
